swarm/pss/pss_test.go

```diff
@@ -517,24 +524,26 @@ func testNetwork(t *testing.T) {
 
 	trigger := make(chan discover.NodeID)
 
-	var adapter adapters.NodeAdapter
-	if paramstring[4] == "exec" {
+	var a adapters.NodeAdapter
+	if adapter == "exec" {
 		dirname, err := ioutil.TempDir(".", "")
 		if err != nil {
 			t.Fatal(err)
 		}
-		adapter = adapters.NewExecAdapter(dirname)
-	} else if paramstring[4] == "sock" {
-		adapter = adapters.NewSocketAdapter(services)
-	} else {
-		adapter = adapters.NewSocketAdapter(services)
+		a = adapters.NewExecAdapter(dirname)
+	} else if adapter == "sock" {
+		a = adapters.NewSocketAdapter(services)
+	} else if adapter == "sim" {
+		a = adapters.NewSimAdapter(services)
 	}
-	net := simulations.NewNetwork(adapter, &simulations.NetworkConfig{
+	net := simulations.NewNetwork(a, &simulations.NetworkConfig{
 		ID: "0",
 	})
 	defer net.Shutdown()
 
-	f, err := os.Open(fmt.Sprintf("testdata/snapshot_%s.json", paramstring[1]))
+	time.Sleep(100 * time.Millisecond)
```

Comment:
why this wait?

Reply:
I noticed that sometimes messages are sent prior to the network being setup. Therefore I added the delay, so that I am sure the cluster is up, prior to sending messages. Obviously it'd be better if there is a channel for that, but as a short-term fix it did the trick.

If you remove the delay, the test will start failing for `sock`.

Reply:
Actually you're right - this `Sleep` is redundant, but the one after `net.Load` is necessary, unless there is a mechanism to wait for the clients setup to be done.